xsd-parser/src/lib.rs

```diff
@@ -1,8 +1,5 @@
 #[cfg(test)]
 #[macro_use]
-extern crate log;
-#[cfg(test)]
-#[macro_use]
```

Comment:
I see quite a few different logical changes in the same commit. I understand that you probably needed most of these changes to satisfy clippy (which is being added to CI here) but they would be separate changes, still. Not something we can fix with fixup so just for future reference. :)